common/es/src/main/java/io/apiman/common/es/util/DefaultEsClientFactory.java

```diff
@@ -92,12 +92,12 @@ public RestHighLevelClient createClient(Map<String, String> config, String defau
      */
     private RestHighLevelClient createEsClient(Map<String, String> config, String indexNamePrefix, List<String> defaultIndices) {
         String host = config.get("client.host"); //$NON-NLS-1$
-        Integer port = NumberUtils.toInt(config.get("client.port"), 9200); //$NON-NLS-1$
+        int port = NumberUtils.toInt(config.get("client.port")); //$NON-NLS-1$
```

Comment:
Is there a reason why you removed the default port?

Reply:
Ah yes, I need to change this back. 

I put it there because the behaviour of `NumberUtils.toInt` was causing confusing behaviour. When a variable substitution from system properties was meant to be happening (but was failing and remaining as text because it had not yet been set), `NumberUtils.toInt` was replacing it with 9200 rather than throwing an error.

i.e. `test.FOO_BAR` was meant to be replaced with the randomised port, but it was staying as `test.FOO_BAR`; that caused a silent error in `NumberUtils.toInt` and it was changing to 9200 rather than throwing an error.

I reckon that if the user provides no input, then a default makes sense. However, if they *do* provide an input but it is erroneous (e.g. text when it should be a number), we should throw an error rather than using a default.

I'll change it back (the alteration was just for testing), but we should open a ticket for that and change it slightly. We can make some convenience functions for this to better behave as we want.

Reply:
#1012 👍